File: ios/SsdidWallet/Domain/Model/VerifiablePresentation.swift
import Foundation

/// W3C Verifiable Presentation wrapper for presenting one or more VCs.
/// Used for JSON-LD VC presentations. For SD-JWT VCs, the presentation
/// is implicit (SD-JWT + selected disclosures + KB-JWT).
struct VerifiablePresentation: Codable {
    let context: [String]
    let type: [String]
    let holder: String
    var verifiableCredential: [VerifiableCredential]
    var proof: Proof?

    enum CodingKeys: String, CodingKey {
        case context = "@context"
        case type, holder, verifiableCredential, proof
    }

    init(
        context: [String] = ["https://www.w3.org/ns/credentials/v2"],
        type: [String] = ["VerifiablePresentation"],
        holder: String,
        verifiableCredential: [VerifiableCredential] = [],
        proof: Proof? = nil
    ) {
        self.context = context
        self.type = type
        self.holder = holder
        self.verifiableCredential = verifiableCredential
        self.proof = proof
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/model/VerifiablePresentation.kt
package my.ssdid.sdk.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * W3C Verifiable Presentation wrapper for presenting one or more VCs.
 *
 * Used for JSON-LD VC presentations. For SD-JWT VCs, the presentation
 * is implicit (SD-JWT + selected disclosures + KB-JWT).
 */
@Serializable
data class VerifiablePresentation(
    @SerialName("@context") val context: List<String> = listOf("https://www.w3.org/ns/credentials/v2"),
    val type: List<String> = listOf("VerifiablePresentation"),
    val holder: String,
    val verifiableCredential: List<VerifiableCredential> = emptyList(),
    val proof: Proof? = null
)
